arch/arm/src/samv7/sam_spi_slave.c

```diff
@@ -485,7 +486,8 @@ static int spi_interrupt(int irq, void *context, FAR void *arg)
 
           /* Report the receipt of data to the SPI device driver */
 
-          SPI_SDEV_RECEIVE(priv->sdev, data);
+          SPI_SDEV_RECEIVE(priv->sdev, (const uint16_t *)&data,
```

Comment:
should we return error if nwords != 1 for all new function?

Reply:
maybe, since this driver can't buffer the data. But in practice, this implements the same functionality as before, so any slave device currently using this wouldn't assume that the controller driver would buffer data anyhow. I'll leave it to you to decide. I can add an error in here, but I'd rather not modify this file too much since I can't test it.

Reply:
Ok, I am fine without the check.